modules/document_repository/ajax/documentEditUpload.php

```diff
@@ -52,34 +51,37 @@
     || $userSingleton->hasPermission('document_repository_delete')
 ) {
     if ($action == 'upload') {
-        $puser      = $_POST['user'];
-        $category   = $_POST['category'];
-        $site       = $_POST['site']       !== '' ? $_POST['site'] : null;
-        $instrument = $_POST['instrument'] !== '' ? $_POST['instrument'] : null;
-        $pscid      = $_POST['pscid']      !== '' ? $_POST['pscid'] : null;
-        $visit      = $_POST['visit']      !== '' ? $_POST['visit'] : null;
-        $comments   = $_POST['comments']   !== '' ? $_POST['commnets'] : null;
-        $version    = $_POST['version']    !== '' ? $_POST['version'] : null;
-
-        $fileSize = $_FILES["file"]["size"];
-        $fileName = $_FILES["file"]["name"];
-        $fileType = end((explode(".", $fileName)));
-
-        // __DIR__ is the document_repository ajax directory
-        // when this script is executing. Go up a level to the
-        // document_repository module directory, and use a
-        // user_uploads directory as a base for user uploads
-        $base_path = __DIR__ . "/../user_uploads/";
-        $fileBase  = $puser . "/" . $fileName;
-
-        if (!file_exists($base_path . $puser)) {
-            mkdir($base_path . $puser, 0777);
+        $category   = $_POST['category']; // required
+        $site       = $_POST['site'] ?? ''; // required but allowed to be empty
+        $instrument = $_POST['instrument'] ?? null;
+        $pscid      = $_POST['pscid']      ?? null;
+        $visit      = $_POST['visit']      ?? null;
+        $comments   = $_POST['comments']   ?? null;
+        $version    = $_POST['version']    ?? null;
+
+        $fileSize = $_FILES['file']['size'];
+        $fileName = $_FILES['file']['name'];
+        $fileType = pathinfo($_FILES['file']['name'], PATHINFO_EXTENSION);
+
+        $uploadPath = "$base/modules/document_repository/user_uploads/$name/";
+        $fullPath   = $uploadPath . $fileName;
+
+        if (!is_writable($uploadPath)) {
+            if (file_exists($uploadPath)) {
+                error_log("Could not write to $uploadPath. Check permissions");
+                exit;
+            }
+            error_log(
```

Comment:
Is this necessary?

Reply:
I thought it might be good to log when a new folder in the file system is being created but I can remove this if you think it's too verbose.

Reply:
I just don't think anyone would go into error log to check that.
But I think we should have a log directory though to put those kind of things